### ISSUE TITLE
Migration to @react-pdf/renderer v3.0.0 causes build and runtime failures

### ISSUE THREAD

**Issue by SashaShostyr** [diegomura/react-pdf#2028]
**Describe the bug**

After updating to **@react-pdf/renderer** **v3.0.0** build job and runtime is failing with error:
<img width="355" alt="image" src="https://user-images.githubusercontent.com/19342294/191956968-e571adc0-e0bd-443e-a6ff-f65aa817dee6.png">

**To Reproduce**

Steps to reproduce the behavior including code snippet (if applies):
1. Clone this repository https://github.com/SashaShostyr/react-pdf-renderer-issue
2. Go to first commit and run app - PDF is rendered correctly
3. Go to second commit and run app - PDF is not rendered and throw an error (see screenshots)

**Expected behavior**

No errors after update.

**Screenshots**

<img width="355" alt="image" src="https://user-images.githubusercontent.com/19342294/191956968-e571adc0-e0bd-443e-a6ff-f65aa817dee6.png">

**Desktop (please complete the following information):**

 - OS: MacOS
 - Browser: Chrome
 - React-pdf version v3.0.0
 - Node version: v16.16.0

**Comment by carlobeltrame** [comment]
Others have reported the same issues when upgrading. Can you try uninstalling and re-installing @react-pdf/renderer? The dependency `@react-pdf/font` must be at least at version `2.3.0` in your lockfile for @react-pdf/renderer v3.0.0 to work.

**Comment by SashaShostyr** [comment]
@carlobeltrame The same issue with `@react-pdf/font: 2.3.0`. I guess it is related to `fontkit` package.
<img width="390" alt="image" src="https://user-images.githubusercontent.com/19342294/192966405-89ca5cee-b515-4144-8972-b44ed3dfcd62.png">

**Comment by carlobeltrame** [comment]
I see. Did you really uninstall and re-install react-pdf? It looks like you simply installed the font package instead of doing that. I just freshly [installed react-pdf v3.0.0](https://github.com/gloggi/qualix/pull/280/commits/90c3d42fb0018ad1086094e98d8fbfde58007113#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519) in a project this week, and there were no problems whatsoever. And last month I also [updated react-pdf in an existing project](https://github.com/ecamp/ecamp3/pull/2949/files) to v3.0.0, and had no problems with it.
Can you provide a reproduction repository which I can have a look at? Some other people have reported the same issues as you, but nobody so far has been able to provide that, and as long as I can't reproduce the issue it's impossible for me to pin the issue down.

**Comment by SashaShostyr** [comment]
@carlobeltrame Yes, I did fresh install `rm -rf node_modules package-lock.json && npm i`. You can check in this repo https://github.com/SashaShostyr/react-pdf-renderer-issue, I've attached it to issue description as well.

**Comment by carlobeltrame** [comment]
Thanks! From a first look, it looks like this is a problem with Webpack 4, nothing really seems to be wrong with react-pdf. If I [upgrade](https://github.com/SashaShostyr/react-pdf-renderer-issue/pull/1) your sample project to Webpack 5 (according to the razzle instructions [here](https://razzlejs.org/docs/upgrade-guide)), everything works as normal again.

I found out that webpack 4 for some reason does not transpile the `import * as fontkit from 'fontkit'` and other statements in the `@react-pdf/font` package correctly. The `fontkit` variable just contains a URL where the fontkit code can be downloaded, instead of importing that code and providing it as a package as it is supposed to. It's good to know that there might be a general issue with webpack 4. Since webpack 4 is still relatively commonly used, I'll look into workarounds for webpack 4 which we could document here.

**Comment by carlobeltrame** [comment]
Someone in #2015 was quicker than me, and already found the root cause of the issue in webpack 4. If you cannot update to webpack 5, you either have to eject and adjust your config as described https://github.com/waku-org/js-waku/issues/895#issuecomment-1220219004 (or use craco or something similar to do that), or stay on @react-pdf/renderer < 3.0.0 until https://github.com/facebook/create-react-app/pull/12605 or https://github.com/facebook/create-react-app/pull/12021 is merged and released, and then update create-react-app.

**Comment by SashaShostyr** [comment]
@carlobeltrame Thank you! Have you tried this comment https://github.com/waku-org/js-waku/issues/895#issuecomment-1220219004 in repository that I attached? I still can't get it working with webpack v4, so if you had a chance to have a look and procide some working example (in my repository) it would be really cool.

**Comment by carlobeltrame** [comment]
I have tried it for a few hours, but did not manage to get it to work. Among other things, I tried setting the following razzle config:
```js
'use strict';

module.exports = {
  options: {
    buildType: 'single-page-application',
  },
  modifyWebpackConfig ({ env: { target, dev }, webpackConfig, webpackObject, options: { pluginOptions, razzleOptions, webpackOptions }, paths }) {
    webpackConfig['module']['rules'][0].test = /\.(js|mjs|jsx|ts|tsx|cjs)$/
    console.log(webpackConfig['module'])
    return webpackConfig
  },
};
```
But that didn't help. That said, I don't really have much experience with Webpack 4 and React, and I have never heard of razzle before this issue, and don't really have time to get deeply accustomed with them.

I recommend you do one of the following:
* Just upgrade to Webpack 5, it's not that hard and you will have to do it sooner or later anyways
* Keep @react-pdf/renderer at v2.x and keep using Webpack 4
* Research Webpack 4 internals and find a config change that works for @react-pdf/renderer v3.0.0
